src/main/java/sirius/biz/jobs/batch/file/XMLExportJob.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.jobs.batch.file;
+
+import sirius.biz.process.ProcessContext;
+import sirius.biz.storage.layer3.FileOrDirectoryParameter;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.health.Exceptions;
+import sirius.kernel.xml.XMLStructuredOutput;
+import sirius.web.resources.Resource;
+import sirius.web.resources.Resources;
+
+import javax.annotation.Nonnull;
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Provides basic functionalities to write data into XML file entries inside an archive.
+ */
+public abstract class XMLExportJob extends ArchiveExportJob {
+
+    @Part
+    private static Resources resources;
+
+    private final boolean requireValidFile;
+    private final String xsdResourcePath;
+    protected XMLStructuredOutput xml;
+    private OutputStream xmlOutputStream;
+
+    /**
+     * Creates a new job for the given process context.
+     *
+     * @param destinationParameter the parameter used to select the destination for the file being written
+     * @param process              the context in which the process will be executed
+     */
+    protected XMLExportJob(@Nonnull XMLExportJobFactory factory,
+                           FileOrDirectoryParameter destinationParameter,
+                           ProcessContext process) {
+        super(destinationParameter, process);
+        requireValidFile = process.getParameter(factory.requireValidFile).orElse(false);
+        xsdResourcePath = factory.getXsdResourcePath();
+    }
+
+    protected void initializeXmlFile(String filename) throws IOException {
+        closeOpenStream();
+        xmlOutputStream = createEntry(filename);
+        xml = new XMLStructuredOutput(xmlOutputStream);
+    }
+
+    @Override
+    public void close() throws IOException {
+        closeOpenStream();
+        super.close();
```

Comment:
Careful: This may throw an exception, in which case the digestExportedFile wont get executed

Reply:
Yup, but if we can't save the target file then there is not much to validate anyway. Besides, the job will also fail.